### PR TITLE
OCPBUGS-17526: clock class is not set to 140 when DPLL is lost source and out of spec - from holdover

### PR DESCRIPTION
**Pull request by aneeshkp**
Defining HOLDOVER/LOCKED and FREERUN state from DPLL 
With Lost Source, IN Spec, and Out of Spec state transitions.
When GNSS is Lost, if the DPLL is in HOLDover and inSpec then do not set it to FREERUN, Instead, GM should be in Holdover state

### CODE REVIEW

File: pkg/event/event.go [hunk 1]
```diff
@@ -267,6 +298,10 @@ connect:
 
 	glog.Info("starting grandmaster state monitoring...")
 	lastGmState := PTP_NOTSET
+	select {
+	case readyCh <- true:
```
Comment (jzding) [inline pkg/event/event.go:302]: what does this do?

File: pkg/event/event.go [hunk 2]
```diff
@@ -373,13 +418,15 @@ connect:
...
 
 			// update clock class
-			go func() {
-				select {
-				case clockClassRequestCh <- ClockClassRequest{cfgName: event.CfgName, gmState: gmState, clockType: event.ClockType}:
-				default:
-					glog.Error("clock class request not read")
-				}
-			}()
+			if gmState != lastGmState {
```
Comment (jzding) [inline pkg/event/event.go:421]: this will always be false due to  line 407 set lastGmState = gmState

File: pkg/event/event.go [hunk 3]
```diff
@@ -452,9 +512,13 @@ func (e *EventHandler) updateCLockClass(cfgName string, ptpState PTPState, clock
...
-			// update only when ClockClass is changed
-			if g.ClockQuality.ClockClass != fbprotocol.ClockClass7 {
+			if e.isSourceLost(cfgName) && !e.outOfSpec && g.ClockQuality.ClockClass != fbprotocol.ClockClass7 {
+				g.ClockQuality.ClockClass = fbprotocol.ClockClass7
+				g.ClockQuality.ClockAccuracy = fbprotocol.ClockAccuracyUnknown
+				// T-REC-G.8275.1-202211-I section 6.3.5
+				g.ClockQuality.OffsetScaledLogVariance = 0xffff
+				err = gmSetterFn(cfgName, g)
+			} else if e.isSourceLost(cfgName) && e.outOfSpec && g.ClockQuality.ClockClass != protocol.ClockClassOutOfSpec {
 				g.ClockQuality.ClockClass = fbprotocol.ClockClass7
```
Comment (jzding) [inline pkg/event/event.go:522]: should be 
`g.ClockQuality.ClockClass = protocol.ClockClassOutOfSpec`

File: pkg/event/event.go [hunk 4]
```diff
@@ -420,10 +555,10 @@ func (e *EventHandler) updateCLockClass(cfgName string, ptpState PTPState, clock
 		return err, clockClass
 	}
 	glog.Infof("current GRANDMASTER_SETTINGS_NP:\n%s", g.String())
-	switch ptpState {
-	case PTP_LOCKED:
-		switch clockType {
-		case GM:
+	switch clockType {
+	case GM:
```
Comment (josephdrichard) [inline pkg/event/event.go:559]: Can you add comments here what each clock class is?

File: pkg/event/event.go (hunk 2 above)
Comment (aneeshkp) [reply pkg/event/event.go]: review again pls

File: pkg/event/event.go (hunk 3 above)
Comment (aneeshkp) [reply pkg/event/event.go]: review again pls

File: pkg/event/event.go (hunk 4 above)
Comment (aneeshkp) [reply pkg/event/event.go:576]: added

**Review by jzding — APPROVED** [review]
/lgtm

**Pull request merged**